### DISCUSSION TITLE
How to obtain PR Curves for a Single Class?

### DISCUSSION THREAD

**Discussion by serzaxlucifer** [ultralytics/yolov5#12831] [Q&A]
I have a dataset and I trained YOLOv5x, YOLOv5l, YOLOv5m on this dataset. This dataset contains 8 classes so these models were trained on these 8 classes. I ultimately now want to plot PRcurve graphs of each class separately comparing different models accuracy. So, I want to plot YOLOv5x, YOLOv5l and YOLOv5m's PR Curve for a specific class (Say 0) on a single graph. But the results of these models contain a PR curve for all classes at once. Can I rather get the curve with one class only? (my class of interest that is).

**Comment by jmarintur** [comment]
Hi @serzaxlucifer, you can get the curve for a specific class by filtering all the bounding boxes you get per image, leaving, or only considering, the ones corresponding to the class you want to evaluate the models on. Once you have your test images with only the predicted bounding boxes corresponding to the target class, you can compute the PR for that class. Check that you use for the ground truth only the bounding boxes of the target class too.
